src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,119 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.task.Importance;
+import com.tstepnik.planner.domain.task.Task;
+import com.tstepnik.planner.domain.user.User;
+import com.tstepnik.planner.repository.TaskRepository;
+import com.tstepnik.planner.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.security.Principal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.LITTLE_IMPORTANT;
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        archiveTasks();
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getAllLoggedUserTasks(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUser(loggedUser);
+    }
+
+    public List<Task> getAllLoggedUserTasksOrderByImportance(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserOrderByImportance(loggedUser);
+    }
+
+    public List<Task> getLoggedUserParticularTasks(Principal principal, Importance importance) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserIdaAndImportance(loggedUser.getId(), importance);
+    }
+
+    public List<Task> getUserTasks(Long userId) {
+        archiveTasks();
+        return taskRepository.findALLByUserId(userId);
+    }
+
+    public Task getTaskById(Long taskId) {
+        archiveTasks();
+        Optional<Task> task = taskRepository.findById(taskId);
+        return task.get();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setCreationDate(LocalDateTime.now());
+        task.setExpirationDate(endOfTheDay());
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
```

Comment:
hmm, i think instead of having User object in task i think we could have only userId. This way we have less tables and do less queries to DB when saving task